Warnings in building ununpack on GCC8

<!-- Before filling this issue, please read the wiki (https://github.com/fossology/fossology/wiki)
and search if the bug do not already exists in the issues (https://github.com/fossology/fossology/issues). -->

### Logs

```lisp
traverse.c: In function ‘TraverseChild’:
traverse.c:140:9: warning: ‘strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
         strncpy(CI->PartnameNew, UploadFileName, sizeof(CI->PartnameNew) - 1);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ununpack-iso.c: In function ‘ExtractISO’:
ununpack-iso.c:145:123: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4019 and 16304 [-Wformat-truncation=]
     snprintf(Cmd,sizeof(Cmd),"(isoinfo -R -J -i '%s' -x '%s' || isoinfo -R -i '%s' -x '%s' || isoinfo -i '%s' -x '%s') > '%s' 2>/dev/null",TempSource,Line+Len,TempSource,Line+Len,TempSource, Line+Len,Line);
                                                                                                                           ^~                                                                            ~~~~
ununpack-iso.c:145:5: note: ‘snprintf’ output 94 or more bytes (assuming 16474) into a destination of size 16384
     snprintf(Cmd,sizeof(Cmd),"(isoinfo -R -J -i '%s' -x '%s' || isoinfo -R -i '%s' -x '%s' || isoinfo -i '%s' -x '%s') > '%s' 2>/dev/null",TempSource,Line+Len,TempSource,Line+Len,TempSource, Line+Len,Line);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ununpack-disk.c: In function ‘ExtractDisk’:
ununpack-disk.c:477:60: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4078 and 16363 [-Wformat-truncation=]
     snprintf(Cmd,sizeof(Cmd),"icat -f '%s' '%s' '%s' > '%s/%s.deleted.%s' 2>/dev/null",
                                                            ^~
         FStype,TempSource,TempInode,TempDest,TempS,TempInode);
                                              ~~~~~          
ununpack-disk.c:477:5: note: ‘snprintf’ output 44 or more bytes (assuming 20519) into a destination of size 16384
     snprintf(Cmd,sizeof(Cmd),"icat -f '%s' '%s' '%s' > '%s/%s.deleted.%s' 2>/dev/null",
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         FStype,TempSource,TempInode,TempDest,TempS,TempInode);
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ununpack-disk.c:415:60: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4078 and 16363 [-Wformat-truncation=]
     snprintf(Cmd,sizeof(Cmd),"icat -f '%s' '%s' '%s' > '%s/%s' 2>/dev/null",
                                                            ^~
         FStype,TempSource,TempInode,TempDest,TempS);
                                              ~~~~~          
ununpack-disk.c:415:5: note: ‘snprintf’ output 35 or more bytes (assuming 16415) into a destination of size 16384
     snprintf(Cmd,sizeof(Cmd),"icat -f '%s' '%s' '%s' > '%s/%s' 2>/dev/null",
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         FStype,TempSource,TempInode,TempDest,TempS);
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
utils.c: In function ‘IsInflatedFile’:
utils.c:56:3: warning: ‘strncpy’ specified bound 4096 equals destination size [-Wstringop-truncation]
   strncpy(FileNameParent, FileName, sizeof(FileNameParent));
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
utils.c: In function ‘RunCommand’:
utils.c:699:44: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4093 and 12283 [-Wformat-truncation=]
     snprintf(Cmd1,sizeof(Cmd1),"%s %s '%s' %s",
                                            ^~
         Cmd,TempPre,TempFile,TempPost);
                              ~~~~~~~~       
utils.c:699:5: note: ‘snprintf’ output 6 or more bytes (assuming 12291) into a destination of size 12288
     snprintf(Cmd1,sizeof(Cmd1),"%s %s '%s' %s",
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         Cmd,TempPre,TempFile,TempPost);
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
utils.c:694:43: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4094 and 12284 [-Wformat-truncation=]
     snprintf(Cmd1,sizeof(Cmd1),"%s %s '%s/%s' %s",
                                           ^~
         Cmd,TempPre,TempCwd,TempFile,TempPost);
                             ~~~~~~~~       
utils.c:694:5: note: ‘snprintf’ output 7 or more bytes (assuming 16387) into a destination of size 12288
     snprintf(Cmd1,sizeof(Cmd1),"%s %s '%s/%s' %s",
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         Cmd,TempPre,TempCwd,TempFile,TempPost);
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/ununpack/agent/traverse.c
+++ b/src/ununpack/agent/traverse.c
@@ -137,7 +137,8 @@ void	TraverseChild	(int Index, ContainerInfo *CI, char *NewDir)
       if (CI->TopContainer && UseRepository)
       {
         RemovePostfix(UploadFileName);
-        strncpy(CI->PartnameNew, UploadFileName, sizeof(CI->PartnameNew) - 1);
+        memcpy(CI->PartnameNew, UploadFileName, sizeof(CI->PartnameNew));
+	CI->PartnameNew[sizeof(CI->PartnameNew)-1] = 0;
       }
       else
         /**

--- a/src/ununpack/agent/traverse.c
+++ b/src/ununpack/agent/traverse.c
@@ -138,7 +138,7 @@ void	TraverseChild	(int Index, ContainerInfo *CI, char *NewDir)
       {
         RemovePostfix(UploadFileName);
         memcpy(CI->PartnameNew, UploadFileName, sizeof(CI->PartnameNew));
-	CI->PartnameNew[sizeof(CI->PartnameNew)-1] = 0;
+        CI->PartnameNew[sizeof(CI->PartnameNew)-1] = 0;
       }
       else
         /**

--- a/src/ununpack/agent/ununpack-disk.c
+++ b/src/ununpack/agent/ununpack-disk.c
@@ -299,7 +299,7 @@ permlist *	SetDiskPerm	(char *inode, permlist *List,
 int	ExtractDisk	(char *Source, char *FStype, char *Destination)
 {
   int rc;
-  char Cmd[FILENAME_MAX*4]; /* command to run */
+  char Cmd[FILENAME_MAX*7]; /* command to run */
   char Line[FILENAME_MAX*2];
   char *s;
   FILE *Fin;

--- a/src/ununpack/agent/ununpack-iso.c
+++ b/src/ununpack/agent/ununpack-iso.c
@@ -78,7 +78,7 @@ mode_t	GetISOMode	(char *Line)
  **/
 int	ExtractISO	(char *Source, char *Destination)
 {
-  char Cmd[FILENAME_MAX*4]; /* command to run */
+  char Cmd[FILENAME_MAX*5]; /* command to run */
   char Line[FILENAME_MAX];
   int Len;
   char *s; /* generic string pointer */

--- a/src/ununpack/agent/utils.c
+++ b/src/ununpack/agent/utils.c
@@ -51,9 +51,9 @@ int IsInflatedFile(char *FileName, int InflateSize)
 {
   int result = 0;
   char FileNameParent[PATH_MAX];
-  memset(FileNameParent, 0, PATH_MAX);
   struct stat st, stParent;
-  strncpy(FileNameParent, FileName, sizeof(FileNameParent));
+  memcpy(FileNameParent, FileName, sizeof(FileNameParent));
+  FileNameParent[PATH_MAX-1] = 0;
   char  *lastSlashPos = strrchr(FileNameParent, '/');
   if (NULL != lastSlashPos)
   {

--- a/src/ununpack/agent/utils.c
+++ b/src/ununpack/agent/utils.c
@@ -630,7 +630,7 @@ void	CheckCommands	(int Show)
 int	RunCommand	(char *Cmd, char *CmdPre, char *File, char *CmdPost,
     char *Out, char *Where)
 {
-  char Cmd1[FILENAME_MAX * 3];
+  char Cmd1[FILENAME_MAX * 5];
   char CWD[FILENAME_MAX];
   int rc;
   char TempPre[FILENAME_MAX];

